cadence/freshmint-encoding/FreshmintEncoding.cdc

```diff
@@ -0,0 +1,105 @@
+/// FreshmintEncoding is a set of utilities for encoding Cadence values as byte arrays.
+///
+/// Variable-legnth values (String, Int, UInt) include a fixed-size length prefix to
+/// prevent distinct sets of values from encoding to the same byte sequence when concatenated.
+///
+/// For example, for a structure with two fields:
+///
+///  - foo: String
+///  - bar: String
+///
+/// Without a length prefix, these instances would produce the same encoding:
+///
+///  - Instance A: { foo: "foo", bar: "bar" }
+///  - Instance b: { foo: "foob", bar: "ar" }
+///
+/// when using an encoding function like this:
+/// 
+/// let encoding = encodeString(foo).concat(encodeString(bar))
+///
+pub contract FreshmintEncoding {
+
+    pub fun encodeUInt(_ value: UInt): [UInt8] {
+        let bytes = value.toBigEndianBytes()
+        let length = UInt16(bytes.length)
+
+        // Prefix byte encoding with a fixed-size encoding of its length
+        return length.toBigEndianBytes().concat(bytes)
```

Comment:
Is there a way to make sure `length` is encoded over a constant size? 
For instance if the `length = 255` or `length = 256`, will the encoding be the same size in bytes?